### PR TITLE
Fix publishing to maven local not working

### PR DESCRIPTION
**Pull request by Insprill**
# Description
Fixes Gradle trying to sign jars locally by checking if the `CI` env var exists, rather than `JITPACK`. ([source](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables))

This also makes it so the version is the git revision prefixed with `dev-` when built locally, since the `mockbukkit.version` property only exists when running in the GitHub action.

# Checklist
The following items should be checked before the pull request can be merged.
- [ ] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -82,7 +82,7 @@ artifacts {
 def isFork = isFork()
 
 signing {
-    required { !isFork && System.getenv('JITPACK') == null }
+    required { !isFork && System.getenv('CI') != null }
```
Comment (TheBusyBiscuit) [inline build.gradle:85]: What about jitpack compatibility though?
This would break that, wouldn't it?

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (Insprill) [reply build.gradle:85]: It shouldn't, no. Before, it wouldn't sign if `JITPACK` was present. Now, it won't sign if `CI` *isn't* present, which anywhere outside of GitHub actions, its not.

Comment (TheBusyBiscuit) [reply build.gradle:85]: Alright

**Review by TheBusyBiscuit — APPROVED** [review]
lgtm

**Pull request merged**